### PR TITLE
add gen_rand_rpi for pi circuit

### PR DESCRIPTION
**Pull request by pinkiebell**
Add a `gen_rand_rpi` function to commit to the `raw_public_inputs_col` values used in Pi circuit.

### CODE REVIEW

File: zkevm-circuits/src/pi_circuit.rs [hunk 1]
```diff
@@ -995,6 +1013,28 @@ fn raw_public_inputs_col<F: Field, const MAX_TXS: usize, const MAX_CALLDATA: usi
...
+pub fn gen_rand_rpi<F: Field, const MAX_TXS: usize, const MAX_CALLDATA: usize>(
+    public_data: &PublicData,
+) -> F {
+    let rlc_rpi_col = raw_public_inputs_col::<F, MAX_TXS, MAX_CALLDATA>(public_data, F::default());
+    let mut keccak = Keccak::default();
+    let mut buf: Vec<u8> = Vec::with_capacity(rlc_rpi_col.len() * 32);
+    for value in rlc_rpi_col.iter() {
+        let mut tmp = value.to_repr();
+        tmp.reverse();
+        // XXX: using keccak.digest() with incremental update yields a different result
```
Comment (pinkiebell) [inline zkevm-circuits/src/pi_circuit.rs:1026]: https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/922

**Comment by pinkiebell** [comment]
@ed255 I created this PR even if this conflicts with #730 just for tracking purposes. If the circuit optimizations gets merged, then I can reflect these changes for the evm contracts. Let me know how to proceed here 👌

**Comment by ed255** [comment]
> @ed255 I created this PR even if this conflicts with #730 just for tracking purposes. If the circuit optimizations gets merged, then I can reflect these changes for the evm contracts. Let me know how to proceed here ok_hand

Sorry I missed this message.

https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/730 has been merged!
I see that it included the `call_data_gas` (now called `call_data_gas_cost`).  So I think you can update this PR accordingly

**Review by ed255 — APPROVED** [review]
LGTM!

**Pull request merged**